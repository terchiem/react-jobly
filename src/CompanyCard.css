.CompanyCard {
  position: relative;
  margin: 10px auto;
  padding: 10px 20px;
  border: 1px solid #555;
  box-shadow: 1px 1px 1px 1px rgba(0, 0, 0, 0.2);
  background-color: #fff;
  text-decoration: none;
}

.CompanyCard-link {
  text-decoration: none;
  color: #111;
}

.CompanyCard-link:hover {
  color: #2469ff;
}

.CompanyCard img {
  position: absolute;
  max-width: 64px;
  top: 0;
  right: 0;
  margin: 10px;
}