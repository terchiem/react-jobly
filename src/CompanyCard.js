import React from 'react';
import { Link } from 'react-router-dom';
import defaultLogo from './defaultLogo.png';
import './CompanyCard.css';

/** Displays basic info about a company */
function CompanyCard({ name, handle, description, logoUrl }) {
  return (
    <Link to={`/companies/${handle}`} className="CompanyCard-link">
      <div className="CompanyCard">
        <div>
          <h3>{name}</h3>
          <p>{description}</p>
          <img src={logoUrl || defaultLogo} alt={`${name} Logo`}/>
        </div>
      </div>
    </Link>
  )
}

export default CompanyCard;