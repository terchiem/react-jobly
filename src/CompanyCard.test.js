import React from 'react';
import { MemoryRouter } from 'react-router-dom';
import { render } from '@testing-library/react';
import CompanyCard from './CompanyCard';

test('renders without crashing', () => {
  render(
    <MemoryRouter>
      <CompanyCard />
    </MemoryRouter>
  );
});

test('matches snapshot', () => {
  const { asFragment } = render(
    <MemoryRouter>
      <CompanyCard />
    </MemoryRouter>
  );
  
  expect(asFragment()).toMatchSnapshot();
})