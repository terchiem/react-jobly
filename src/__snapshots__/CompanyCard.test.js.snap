// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <a
    class="CompanyCard-link"
    href="/companies/undefined"
  >
    <div
      class="Card CompanyCard"
    >
      <h3 />
      <p />
      <img
        alt="undefined Logo"
        src="defaultLogo.png"
      />
    </div>
  </a>
</DocumentFragment>
`;
